src/api.ts

```diff
@@ -25,3 +26,34 @@ function get<T>(url: string): Promise<T> {
 export const getTodos = () => get<Todo[]>('/todos');
 
 export const getUser = (userId: number) => get<User>(`/users/${userId}`);
+
+export const filterTodo = (
+  todos: Todo[],
+  filterParam: Filter,
+  queryParam: string,
+) => {
+  let todosCopy = [...todos];
+
+  switch (filterParam) {
+    case Filter.Active:
+      todosCopy = todosCopy.filter(todo => !todo.completed);
+      break;
+    case Filter.Completed:
+      todosCopy = todosCopy.filter(todo => todo.completed);
+      break;
+    case Filter.All:
+    default:
+      break;
+  }
```

Comment:
Use ternary oparetor to avoid reasign value. For example 
`const visibleTodos = filterParam !== Filter.All ? doFilter() : todos`